src/manual_wrappers.jl

```diff
@@ -24,12 +24,36 @@ Return a Julia matrix wrapping the data of a gsl_matrix
 """
 @inline function wrap_gsl_matrix(m::Ptr{gsl_matrix})
     M = unsafe_load(m)
-    @assert M.size2==M.tda "Cannot unsafe_wrap gsl_matrix with tda != size2."     
+    @assert M.size2==M.tda "Cannot unsafe_wrap gsl_matrix with tda != size2."
     return unsafe_wrap(Array{Float64}, M.data, (M.size1, M.size2))
 end
 
 
 ## Root finding
+gsl_function_helper(x::Cdouble, (f,))::Cdouble = f(x)
+
+# The following code relies on `gsl_function` being a mutable type
+# (such that we can call `pointer_from_objref` on it) to simplify the object structure
+# a little bit and avoid hitting some limitation of the allocation optimizer.
+@assert ismutable(gsl_function(C_NULL, C_NULL))
+
+function Base.cconvert(::Type{Ref{gsl_function}}, t::T) where {F,T<:Tuple{F}}
```

Comment:
Please don't. This tuple is totally unnecessary and it also create more opportunity for allocation.